includes/class-wp-job-manager-post-types.php

```diff
@@ -389,14 +390,29 @@ public function register_post_types() {
 		 */
 		add_feed( self::get_job_feed_name(), [ $this, 'job_feed' ] );
 
+		/**
+		 * This code checks if (1) we're on a search or (2) the option to hide expired job listings is enabled.
+		 * If either condition is false, we're going to set the register_post_status() 'public' arg to false,
+		 * otherwise it will be true.
+		 *
+		 * This will prevent single expired job listings from showing a '404', instead of an
+		 * 'Expired' notice, when viewing the single job listing and not logged in.
+		 *
+		 * This will also address historical issues stemming from addressing the issue raised here
+		 * https://github.com/Automattic/WP-Job-Manager/issues/1884.
+		 */
+		$is_site_search        = ( isset( $_GET['s'] ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$is_expired_searchable = (bool) get_option( 'job_manager_hide_expired' );
```

Comment:
This is `true` if we're to hide expired listings, so not searchable; inverse of what the variable name implies. Is that correct? I maybe need more coffee.

Reply:
Committed here https://github.com/Automattic/WP-Job-Manager/pull/2449/commits/d23b3a3242ab91c54286d4fccf57cd5b725ecb43.

I've renamed the variable from `$is_expired_searchable` to `$is_expired_hidden`. That's more readable to my brain, but let me know if it's still a bit muddy.